packages/amplication-server/src/core/build/build.service.ts

```diff
@@ -491,8 +491,9 @@ export class BuildService {
             pullRequestMode,
           };
 
-          await this.queueService.emitMessage(
+          await this.queueService.emitMessageWithKey(
             this.configService.get(Env.CREATE_PR_REQUEST_TOPIC),
+            resourceRepository.id,
             JSON.stringify(createPullRequestArgs)
```

Comment:
There is now a library for standardise the approach. Try to use that instead of code in each service.
`@amplication/util/nestjs/kafka`

Reply:
I've checked util kafka library, I didn't find appropriate service/method. Can you, please, send a link to example of it's usage?

Reply:
As per best practices, instructions on how to use a library are in the library README https://github.com/amplication/amplication/blob/master/libs/util/nestjs/kafka/README.md

Reply:
because we never used this library before it has to be done and tested separately. I've opened a new issue for that:
https://github.com/amplication/amplication/issues/5287

Reply:
That was exactly the point of using the library. Since you implemented a new function that is not covered by any unit test, please try to either use the existing one or create unit tests for this.  
Another option is to update the library with any new implementation so we are not making it obsolete.

Reply:
But it's not a new usage. Replacement of queue service for kafka producer service will affect other flows in build service too and not only. There should be only one producer per service. So I see two options here. The first option is to add another producer form lib nestjs/kafka just for this specific case and there're going to be two producers in server. The second is to replace queue service with kafka producer service in server, which will affect other flows. What should I do?

Reply:
But it's not a new usage. Replacement of queue service for kafka producer service will affect other flows in build service too and not only. There should be only one producer per service. So I see two options here. The first option is to add another producer form lib nestjs/kafka just for this specific case and there're going to be two producers in server. The second is to replace queue service with kafka producer service in server, which will affect other flows. What should I do?

Reply:
Ok then, let's go with this and standardise with the lib later